File: Hotelaria/src/main/java/com/dolts/hotelaria/models/ImagemQuartoModel.java
package com.dolts.hotelaria.models;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.Type;

import java.io.Serializable;

@Entity
@Table(name = "imagens")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
public class ImagemQuartoModel implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nome;
    private String formato;
    private Long tamanho;
    private Long idDoQuartoDaImagem;

    @Lob
    private byte[] imagem;

    @Version
    private Long version;
}

File: hotelaria-frontend/src/app/shared/models/imagemQuarto.model.ts
export class ImagemQuartoModel {
    id: number;
    nome: string;
    formato: string;
    tamanho: number;
    idDoQuartoDaImagem: number;
    imagem: string | File | ArrayBuffer | Blob;
    arquivo: File | Blob;
    version: number;
}
